lib/graphql/graphql-client.ts

```diff
@@ -0,0 +1,46 @@
+import axios, { AxiosRequestConfig, AxiosResponse } from 'axios'
+
+import { GraphQLResponse } from './graphql-schemas'
+
+/* Interface for accessing any GraphQL API */
+export interface IGraphQLClient {
+  fetchData<T>(query: string, variables?: { [key: string]: any }): Promise<T>
+}
+
+/* Implementation of the IGraphQLClient interface to give access to any GraphQL API */
+export class GraphQLClient implements IGraphQLClient {
+  private readonly endpoint: string
+  private readonly headers: { [key: string]: string }
+
+  constructor(endpoint: string, headers: { [key: string]: string }) {
+    this.endpoint = endpoint
+    this.headers = headers
+  }
```

Comment:
style suggestion, we can simplify this with the syntactic sugar:

```
constructor(
  private readonly endpoint: string,
  private readonly headers: Record<string, string>
) {}
```

I'll leave this decision up to you though, whatever seems more readable in your opinion